nixos/services/haproxy/README.md

```diff
@@ -0,0 +1,220 @@
+The HAproxy Web Proxy is enabled when the `webgateway` role is enabled and can
+also be enabled through a custom Nix Configuration Option (see :ref:`nixos-custom-modules`).
+
+If Haproxy is enabled it can be configured by adding
+:file:`/etc/local/haproxy/*.conf` files in the HAproxy format as well as
+from custom Nix Configuration Options.
```

Comment:
explain how to enable structured config or disable plain config, what's the default, and what happens if both are enabled